define.js

```diff
@@ -1000,6 +1009,16 @@ define.setup = function(props, sealed) {
 	if(canReflect.size(instanceDefinitions) > 0) {
 		defineConfigurableAndNotEnumerable(this, "_instanceDefinitions", instanceDefinitions);
 	}
+	if(requiredButNotProvided.size) {
+		var msg, missingProps = Array.from(requiredButNotProvided);
+		if(requiredButNotProvided.size === 1) {
+			msg = `Missing required property [${missingProps[0]}].`;
```

Comment:
Can we add the name of the type to this error message?

Reply:
I'm not sure that we should, the `required` property is unrelated to the type. A property can be required without having an explicit type. I think adding the type might make it confusing, but I also see how it *might* be useful.

Reply:
Sorry, I meant the class, like `canReflect.getName(this)`.

Reply:
Oh I see, so if you are rendering a template with many components you can figure out which one it throwing the error? Makes sense. Probably should check for `localName` possibly even.